### PR TITLE
Clean up PR

### PR DESCRIPTION
**Pull request by nistala**
- fixes frontend merge conflicts and moves dashboard forward from previous PRs
- fixes auth error messages disappearing from previously merged auth PR
- backend for updating task worker result still pending.

### CODE REVIEW

File: crowdsourcing/migrations/0026_taskworker_status.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        ('crowdsourcing', '0025_auto_20150805_1718'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='taskworker',
+            name='status',
+            field=models.IntegerField(default=1, choices=[(1, b'Created'), (2, b'In Progress'), (3, b'Submitted'), (4, b'Accepted'), (5, b'Returned')]),
+        ),
+    ]
```
Comment (dmorina) [inline crowdsourcing/migrations/0026_taskworker_status.py]: this migration is not valid anymore please remove

File: crowdsourcing/migrations/0027_auto_20150807_1736.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        ('crowdsourcing', '0026_taskworker_status'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='taskworker',
+            name='status',
+            field=models.IntegerField(default=1, choices=[(1, b'In Progress'), (2, b'Submitted'), (3, b'Accepted'), (4, b'Returned')]),
+        ),
+    ]
```
Comment (dmorina) [inline crowdsourcing/migrations/0027_auto_20150807_1736.py]: this migration is not valid anymore please remove

File: static/js/crowdsource.routes.js [hunk 3]
```diff
@@ -64,6 +64,14 @@
       controller: 'taskWorkerDetailController',
       controllerAs: 'taskWorkerDetail'
     })
+    .when('/monitor/:moduleId', {
```
Comment (dmorina) [inline static/js/crowdsource.routes.js]: monitor is outdated now since a while

File: static/js/task/controllers/task.controller.js [hunk 4]
```diff
@@ -74,23 +74,15 @@
...
         function skip() {
             Task.skipTask(self.task_id).then(function success(data, status) {
-                    if (data[1]==200){
-                        $location.path('/task/' + data[0].task);
-                    }
-                    else {
-                        $mdToast.showSimple('No tasks left.');
-                    }
-
+                    $location.path('/task/' + data[0].task);
```
Comment (dmorina) [inline static/js/task/controllers/task.controller.js]: this breaks the thing when no tasks left

File: static/js/task/controllers/task.controller.js [hunk 5]
```diff
@@ -104,34 +96,29 @@
...
-                Task.submitTask(requestData).then(
-                    function success(data, status) {
-                        if (task_status == 1 || data[1]!=200) $location.path('/');
-                        else if (task_status == 2)
-                            $location.path('/task/' + data[0].task);
-                    },
-                    function error(data, status) {
-                        $mdToast.showSimple('Could not submit task.');
-                    }).finally(function () {
+            if (self.task_worker_id) {
```
Comment (dmorina) [inline static/js/task/controllers/task.controller.js]: status is important

File: static/js/task/services/task.service.js [hunk 6]
```diff
@@ -73,7 +75,26 @@
       return HttpService.doRequest(settings);
     }
 
-    function submitTask(data){
+    function getSavedTask(task_worker_id) {
+      var settings = {
+        url: '/api/task-worker/' + task_worker_id + '/retrieve_with_data_and_results/?task_worker_id=' + task_worker_id,
+        method: 'GET',
+        params: {
+          id: task_worker_id
```
Comment (dmorina) [inline static/js/task/services/task.service.js]: why do we need to pass task_worker_id three times?

**Comment by dmorina** [comment]
breaks many things for now, needs re-iteration, remove old code

File: crowdsourcing/migrations/0027_auto_20150807_1736.py (hunk 2 above)
Comment (dmorina) [inline crowdsourcing/migrations/0027_auto_20150807_1736.py]: these migrations are not up to date

File: static/js/crowdsource.routes.js [hunk 7]
```diff
@@ -64,6 +64,14 @@
...
+      templateUrl: '/static/templates/task/monitor.html',
+      controller: 'MonitorController'
+    })
+
+    .when('/ImageLabel', {
+      templateUrl: '/static/templates/task/ImageLabel.html'
+    })
```
Comment (dmorina) [inline static/js/crowdsource.routes.js]: remove unused routes

File: static/js/monitor/controllers/monitor.controller.js [hunk 8]
```diff
@@ -152,9 +172,9 @@
...
       var csvString = csvArr.join("%0A");
       var a         = document.createElement('a');
-      a.href        = 'data:attachment/csv,' + csvString;
+      a.href        = 'data:text/csv;charset=utf-8,' + csvString;
       a.target      = '_blank';
-      a.download    = 'data.csv';
+      a.download    = vm.projectName.replace(/\s/g,'') + '_' + vm.moduleName.replace(/\s/g,'') + '_data.csv';
 
       document.body.appendChild(a);
       a.click();
```
Comment (dmorina) [inline static/js/monitor/controllers/monitor.controller.js]: monitor is outdated

File: static/js/task/controllers/task.controller.js [hunk 9]
```diff
@@ -74,23 +74,15 @@
 
         function skip() {
             Task.skipTask(self.task_id).then(function success(data, status) {
-                    if (data[1]==200){
```
Comment (dmorina) [inline static/js/task/controllers/task.controller.js]: we need http status here

File: static/js/task/controllers/task.controller.js [hunk 10]
```diff
@@ -104,34 +96,29 @@
...
-                    function error(data, status) {
-                        $mdToast.showSimple('Could not save task.');
-                    }).finally(function () {
 
-                    }
-                );
-            } else {
-                Task.submitTask(requestData).then(
-                    function success(data, status) {
-                        if (task_status == 1 || data[1]!=200) $location.path('/');
```
Comment (dmorina) [inline static/js/task/controllers/task.controller.js]: breaks no tasks left

File: static/js/task/services/task.service.js [hunk 11]
```diff
@@ -73,7 +75,26 @@
       return HttpService.doRequest(settings);
     }
 
-    function submitTask(data){
+    function getSavedTask(task_worker_id) {
+      var settings = {
+        url: '/api/task-worker/' + task_worker_id + '/retrieve_with_data_and_results/?task_worker_id=' + task_worker_id,
```
Comment (dmorina) [inline static/js/task/services/task.service.js]: passing task_worker_id only once should be sufficient

**Comment by dmorina** [comment]
reverted because it was breaking many things

File: static/js/task/controllers/task.controller.js (hunk 9 above)
Comment (nistala) [inline static/js/task/controllers/task.controller.js]: @dmorina auth PR broke error reporting and the upload bug, this is not the right way to handle this use but I added it back.

File: static/js/task/controllers/task.controller.js (hunk 10 above)
Comment (nistala) [inline static/js/task/controllers/task.controller.js]: see comment above - again not the right way to handle this.

File: static/js/task/services/task.service.js (hunk 11 above)
Comment (nistala) [inline static/js/task/services/task.service.js]: backend wasn't accepting it when i tried state may have changed.

**Comment by nistala** [comment]
looks like changes got dragged in from @aginzberg's branch, i've reverted all backend.

File: static/js/task/controllers/task.controller.js (hunk 9 above)
Comment (dmorina) [inline static/js/task/controllers/task.controller.js]: in what way did it break it?
well you could go about status or by data with a message.
it's a SUCCESS status as the task is skipped successfully and if there is no other left you get 204 else 200

File: static/js/task/controllers/task.controller.js (hunk 10 above)
Comment (dmorina) [inline static/js/task/controllers/task.controller.js]: what is the right way to handle this?

**Pull request merged**